Sources/Architecture/X86/HardwareBreakpointManager.cpp

```diff
@@ -48,17 +48,33 @@ ErrorCode HardwareBreakpointManager::add(Address const &address, Type type,
   return super::add(address, type, size, mode);
 }
 
+ErrorCode HardwareBreakpointManager::remove(Address const &address) {
+  auto loc = std::find(_locations.begin(), _locations.end(), address);
+  if (loc != _locations.end()) {
+    _locations[loc - _locations.begin()] = 0;
+  }
+
+  return super::remove(address);
```

Comment:
This won't work, because a watchpoint doesn't get added to _locations until enabled. Which means there is a case where it could be set up in the superclass, but hasn't yet been enabled. So it needs to be removed from the superclass, but it doesn't need to be removed from _locations.

This is evidenced by the number of unit tests that fail if I make this change.